src/test/java/de/bonndan/nivio/appuser/AppUserRepositoryTest.java

```diff
@@ -0,0 +1,46 @@
+package de.bonndan.nivio.appuser;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@DataJpaTest
+class AppUserRepositoryTest {
+
+    @Autowired
+    private AppUserRepository appUserRepository;
+
+    AppUserService appUserService = new AppUserService(appUserRepository);
+
+
+    @Test
+    void findByExternalId() {
+        Optional<AppUser> appUser = Optional.of(new AppUser());
+        appUser.get().setExternalId("100");
+        appUser.get().setAlias("login");
+        appUser.get().setAppUserRole(AppUserRole.USER);
+        appUser.get().setId(1L);
+        appUser.get().setIdp("github");
+
+        appUser.ifPresent(user -> appUserRepository.save(user));
```

Comment:
Please comment with given/when/then or that other assert/assess/a... pattern so that steps in the test are clear.